File: Algorithm/fibonacci.ts
function runAlgo(): void {
    const readline = require('readline-sync');
    const ix: number = parseInt(readline.question("Please enter Fibonacci index: "));

    let currentVal: number = 1, previousVal: number = 0;

    for (let counter = 0; counter < ix; counter++) {
        currentVal += previousVal;
        if (counter !== 0) {
            previousVal = currentVal - previousVal;
        }
        process.stdout.write(`-> ${currentVal} `);
    }
    console.log();
}

runAlgo();

File: Algorithm/fibonacci.cs
using System;

class Program {
    static void RunAlgo() {
        Console.Write("Please enter Fibonacci index: ");
        int ix = int.Parse(Console.ReadLine());

        int currentVal = 1, previousVal = 0;

        for (int counter = 0; counter < ix; counter++) {
            currentVal += previousVal;
            if (counter != 0) {
                previousVal = currentVal - previousVal;
            }
            Console.Write($"-> {currentVal} ");
        }
        Console.WriteLine();
    }

    static void Main() {
        RunAlgo();
    }
}

